secp256kfun/src/scalar.rs

```diff
@@ -497,6 +521,27 @@ mod conversion_impls {
                     }
                 }
 
+                impl<S> TryFrom<$t> for Scalar<S, NonZero> {
+                    type Error = ZeroScalar<$t>;
+
+                    fn try_from(value: $t) -> Result<Self, Self::Error> {
+                        // big-endian integer → 32-byte array
+                        let mut bytes = [0u8; 32];
+                        let int_bytes = value.to_be_bytes();
+                        bytes[32 - int_bytes.len() ..].copy_from_slice(&int_bytes);
+                        let scalar = Scalar::<S, Zero>::from_bytes(bytes).unwrap();
+
+                        // Check if value is zero
+                        if value == 0 {
```

Comment:
nit: could check this first for an early return before doing the computation. Maybe negligible, and maybe the compiler would optimize this